pkg/cli/zip_upload_table_dumps.go

```diff
@@ -302,8 +301,27 @@ func processTableDump(
 
 		lines = append(lines, jsonRow)
 		return nil
-	}); err != nil {
-		return err
+	}
+
+	var processErr error
+
+	// Using new TSV parser for create_statements.txt because it has quoted fields containing newlines
+	if strings.HasSuffix(fileName, "crdb_internal.create_statements.txt") {
+		header, iter := makeQuotedTSVIterator(f)
+		processErr = iter(func(cols []string) error {
+			return processFields(header, cols)
+		})
+	} else {
+		// Using the old parser for all the other files
+		header, iter := makeTableIterator(f)
+		processErr = iter(func(row string) error {
+			cols := strings.Split(row, "\t")
+			return processFields(header, cols)
+		})
+	}
```

Comment:
If the `iter`function have consistent signature, we can reduce the duplication here:

```go
iterMaker := makeTableIterator // default
if strings.HasSuffix(fileName, "crdb_internal.create_statements.txt") {
    // special parser
    iterMaker  = makeQuotedTSVIterator
}

header, iter := iterMaker(f)
processErr = iter(func(cols []string) error {
    return processFields(header, cols)
})
```